src/components/faktum/FaktumPeriode.test.tsx

```diff
@@ -71,18 +72,259 @@ describe("FaktumPeriode", () => {
       );
 
       const datepickerFom = screen.getByLabelText(faktumMockData.beskrivendeId + ".fra");
-      const datepickerTom = screen.getByLabelText(faktumMockData.beskrivendeId + ".til");
 
       // To trigger correct behavior as explained here: https://github.com/testing-library/user-event/issues/399#issuecomment-815664027
+      await user.type(datepickerFom, "04.08.2022");
+
+      await waitFor(() => {
+        expect(onchange).toBeCalledTimes(1);
+        expect(onchange).toHaveBeenCalledWith(faktumMockData, { fom: "2022-08-04" });
+      });
+    });
+  });
+
+  describe("When user adds tom date to existing periode answer", () => {
+    test("Should post fom and tom date to server", async () => {
+      const svar = { fom: "2022-08-04" };
+      faktumMockData.svar = svar;
+
+      const user = userEvent.setup();
+      const onchange = jest.fn();
+
+      render(
+        <MockContext>
+          <FaktumPeriode faktum={faktumMockData} onChange={onchange} />
+        </MockContext>
+      );
+
+      const datepickerFom = screen.getByLabelText(
+        faktumMockData.beskrivendeId + ".fra"
+      ) as HTMLInputElement;
+
+      const datepickerTom = screen.getByLabelText(
+        faktumMockData.beskrivendeId + ".til"
+      ) as HTMLInputElement;
+
+      await waitFor(() => {
+        expect(datepickerFom.value).toBe("04.08.2022");
+      });
+
+      await user.type(datepickerTom, "06.08.2022");
+
+      await waitFor(() => {
+        expect(onchange).toBeCalledTimes(1);
+        expect(onchange).toHaveBeenCalledWith(faktumMockData, {
+          fom: "2022-08-04",
+          tom: "2022-08-06",
+        });
+      });
+    });
+  });
+
+  describe("When user selects tom date that is before fom date", () => {
+    test("Should post just fom date to server and clear tom date", async () => {
+      const svar = { fom: "2022-08-04", tom: "2022-09-05" };
+      faktumMockData.svar = svar;
+
+      const user = userEvent.setup();
+
+      render(
+        <MockContext>
+          <FaktumPeriode faktum={faktumMockData} />
+        </MockContext>
+      );
+
+      const datepickerFom = screen.getByLabelText(
+        faktumMockData.beskrivendeId + ".fra"
+      ) as HTMLInputElement;
+
+      const datepickerTom = screen.getByLabelText(
+        faktumMockData.beskrivendeId + ".til"
+      ) as HTMLInputElement;
+
+      const datePickerError = document.querySelector(
+        ".navds-form-field__error"
+      ) as HTMLInputElement;
+
+      await waitFor(() => {
+        expect(datepickerFom.value).toBe("04.08.2022");
+        expect(datepickerTom.value).toBe("05.09.2022");
+      });
+
+      await user.type(datepickerTom, "05.05.2022");
+
+      await waitFor(() => {
+        expect(datePickerError).toBeInTheDocument();
+      });
+    });
+  });
+
+  describe("When user selects in future date on start date of employment relationship", () => {
```

Comment:
Kanskje referere til hvilken beskrivendeId dette gjelder. :) Og "When user selects in future" -> "When user selects future"